drivers/char/random.c

```diff
@@ -981,7 +981,7 @@ static void xfer_secondary_pool(struct entropy_store *r, size_t nbytes)
 	    r->entropy_count > r->poolinfo->poolfracbits)
 		return;
 
-	if (r->limit == 0 && random_min_urandom_seed) {
+	if (r->limit == 0 && r->initialized && random_min_urandom_seed) {
```

Comment:
This function is called before extracting entropy from a pool, in order to move entropy from the "input" pool into the desired pool.

In order to prevent `/dev/urandom` from starving `/dev/random`, `urandom` will only reseed from the input pool once ever 60s (by default, it's tunable). This conditional block implements that ratelimit.

However, if it hasn't been initialized yet (doesn't have at least 128 bits of entropy), we should still greedily grab entropy until it has been, which is what my change does.  My rationale is that we won't risk long-term starvation, since we'll only need to suck ~128 bits out, ever, and that ensuring the validity of the randomness in `/dev/urandom` ASAP is more important than keeping a few extra bits around to replenish `/dev/urandom`

Which part of that would help you as a comment? :) I don't think that entire graf merits inclusion…